### DISCUSSION TITLE
Is v7 usable?

### DISCUSSION THREAD

**Discussion by mattiaz9** [next-safe-action/next-safe-action#136] [Q&A] [answered]
I'm testing out the beta version but it looks like the type definition has some problems.

Following this example I found in the docs:

```ts
import { createSafeActionClient } from "next-safe-action";
import { z } from "zod";

const actionClient = createSafeActionClient();

const schema = z.object({
  username: z.string().min(3).max(10),
  password: z.string().min(8).max(100),
});

const loginUser = actionClient
  .schema(schema)
  .action(async ({ parsedInput: { username, password } }) => {
    if (username === "johndoe" && password === "123456") {
      return {
        success: "Successfully logged in",
      };
    }

    return { failure: "Incorrect credentials" };
  });
```

The first error is with the schema which is not assignable to `undefined`, unless I add `.metadata(undefined)`.
The second problem is with the `parsedInput` which is always `undefined`.

So my question is if this version is still a work in progress or are those bugs?

**Comment by TheEdoRan** [comment, marked as answer]
Yes, v7 is API stable, just waiting for Next.js 14.3.0 (or 15?) release to publish it to the stable channel.

Regarding this issue, have you installed the right TypeSchema adapter for Zod? 

```sh
npm i @typeschema/zod
```

**Reply by TheEdoRan** [reply]
Updated installation instructions can be found in [this section](https://next.next-safe-action.dev/docs/getting-started#installation) on the website.

**Reply by mattiaz9** [reply]
yup, that was it! I missed that extra package to install
